tests/Unit/Framework/Tests/Test_ActionTesting.cpp

```diff
@@ -542,4 +543,414 @@ SPECTRE_TEST_CASE("Unit.ActionTesting.MockComponent", "[Unit]") {
+  }
+};
+
+struct LocalRank {
+  template <typename MyProxy, typename ArrayIndex>
+  static int f(MyProxy& my_proxy, const ArrayIndex& array_index) noexcept {
+    return sys::my_local_rank(*my_proxy[array_index].ckLocal());
+  }
+};
+
+struct NumProcs {
+  template <typename MyProxy, typename ArrayIndex>
+  static int f(MyProxy& my_proxy, const ArrayIndex& array_index) noexcept {
+    return sys::number_of_procs(*my_proxy[array_index].ckLocal());
+  }
+};
+
+struct NumNodes {
+  template <typename MyProxy, typename ArrayIndex>
+  static int f(MyProxy& my_proxy, const ArrayIndex& array_index) noexcept {
+    return sys::number_of_nodes(*my_proxy[array_index].ckLocal());
+  }
+};
+
+template<int NodeIndex>
+struct ProcsOnNode {
+  template <typename MyProxy, typename ArrayIndex>
+  static int f(MyProxy& my_proxy, const ArrayIndex& array_index) noexcept {
+    return sys::procs_on_node(NodeIndex, *my_proxy[array_index].ckLocal());
+  }
+};
+
+template<int NodeIndex>
+struct FirstProcOnNode {
+  template <typename MyProxy, typename ArrayIndex>
+  static int f(MyProxy& my_proxy, const ArrayIndex& array_index) noexcept {
+    return sys::first_proc_on_node(NodeIndex, *my_proxy[array_index].ckLocal());
+  }
+};
+
+template <int ProcIndex>
+struct NodeOf {
+  template <typename MyProxy, typename ArrayIndex>
+  static int f(MyProxy& my_proxy, const ArrayIndex& array_index) noexcept {
+    return sys::node_of(ProcIndex, *my_proxy[array_index].ckLocal());
+  }
+};
+
+template <int ProcIndex>
+struct LocalRankOf {
+  template <typename MyProxy, typename ArrayIndex>
+  static int f(MyProxy& my_proxy, const ArrayIndex& array_index) noexcept {
+    return sys::local_rank_of(ProcIndex, *my_proxy[array_index].ckLocal());
+  }
+};
+
+template <typename Functor>
+struct ActionSetValueTo {
+  template <typename ParallelComponent, typename DbTagsList,
+            typename Metavariables, typename ArrayIndex,
+            Requires<tmpl::list_contains_v<DbTagsList, ValueTag>> = nullptr>
+  static void apply(db::DataBox<DbTagsList>& box,
+                    Parallel::GlobalCache<Metavariables>& cache,
+                    const ArrayIndex& array_index) noexcept {
+    auto& my_proxy = Parallel::get_parallel_component<ParallelComponent>(cache);
+    auto function_value = Functor::f(my_proxy, array_index);
+    db::mutate<ValueTag>(
+        make_not_null(&box),
+        [&function_value](const gsl::not_null<int*> value) noexcept {
+          *value = function_value;
+        });
+  }
+};
+
+struct MetavariablesOneComponent {
+  using component_list = tmpl::list<ComponentA<MetavariablesOneComponent>>;
+
+  enum class Phase { Initialization, Testing, Exit };
+};
+
+void test_parallel_info_functions() noexcept {
+  using metavars = MetavariablesOneComponent;
+  using component_a = ComponentA<metavars>;
+
+  // Choose 2 nodes with 3 cores on first node and 2 cores on second node.
+  ActionTesting::MockRuntimeSystem<metavars> runner{{},{},{3,2}};
```

Comment:
[nitpick] has this been formatted? I would have expected more space in the initializer list.